lib/local_daily_digest.js

```diff
@@ -0,0 +1,167 @@
+var async = require('async'),
+	_ = require('lodash'),
+	log = require('../log'),
+	Profile = require('../models').Profile,
+	Contact = require('../models').Contact,
+	log = require('../log'),
+	mail = require('../mail');
+
+function get(req, res, next) {
+	var result = {},
+	 	allLocalProfilesWDigest = [],
+	 	allLocalProfiles =[],
+	 	requiredUsers=[],
+	 	checkedOutUsers = [],
+	 	finalCheckedOutList=[],
+		tempManagers =[],
+		tempUsers = [],
+		managerContactsProfile = [],
+		finalOutput = [],
+		tempProf=[],
+		requiredLocationsId =[],
+		requiredLocationsName =[],
+		temp='',
+		tempProfile ='',
+		checkedInUsers = [];
+	
+	async.series([
+		//THe local daily digest will be just the digest for that country !!
+		function (cb) {
+			Profile.find({'dailyDigest': {$not: {$size: 0}}}, function(err, profile){
+			if(err){
+		  		res.send(500, new Error(err));
+		  		return cb(true);
+		  	}
+
+		  	if(!profile){
+		  		res.send(400, new Error('No Profiles were found'));
+		  	}
+
+		  	profile.forEach(function(profile){
+			  		allLocalProfilesWDigest.push(profile);
+			  	});
+	  			return cb();
+			});
+		},
+		function(cb){
+			allLocalProfilesWDigest.forEach(function(profile){
+				// console.log(profile);
+				profile.dailyDigest.forEach(function(location){
+					requiredLocationsId.push(location);
+				});
+				// requiredLocationsName.push(profile.location);
+			});
+			console.log("the countries this profile is either a manger or editor of are: ", requiredLocationsId);
```

Comment:
To be removed, but I don't see any check here or in the save to make sure that the locations added to the local daily digest are only locations which a user is a manager or editor of. I'm not sure this specific check is really needed since at some point the idea is to give access to the daily digest to every user, but we still need to make sure at least that only verified users have access to the daily digest, as currently, an unverified user could subscribe himself to a protected list (ie a list which is supposed to be accessible to verified users only), which is a security issue